Cross-region backups (#4242)
timescale/docs: use-timescale/backup-restore.md

---
title: Back up and recover your Tiger Cloud services
excerpt: See how and when Tiger Cloud backs up your data, making sure you always have something to fall back on in case of disaster recovery
products: [cloud]
keywords: [backups, restore]
tags: [recovery, failures]
---

# Back up and recover your $SERVICE_SHORTs

> $CLOUD_LONG automatically handles backup for your $SERVICE_LONGs using the `pgBackRest` tool. You don't need to perform backups manually. What's more, with [cross-region backup][cross-region], you are protected even if an entire AWS region goes down.

Reviewer: ```suggestion
$CLOUD_LONG automatically handles backup for your $SERVICE_LONGs using the `pgBackRest` tool. You don't need to perform backups manually. What's more, with [cross-region backup][cross-region], you are protected when an entire AWS region goes down.
```

$CLOUD_LONG automatically creates one full backup every week, and

> incremental backups every day in the same region as your $SERVICE_SHORT. Additionally, all WAL ([Write-Ahead Log][wal])

Reviewer: ```suggestion
incremental backups every day in the same region as your $SERVICE_SHORT. Additionally, all [Write-Ahead Log (WAL)][wal]```

files are retained back to the oldest full backup. This means that you always
have a full backup available for the current and previous week: 

![Backup in Tiger Cloud](https://assets.timescale.com/docs/images/database-backup-recovery.png)

> In the event of a storage failure, a $SERVICE_SHORT automatically recovers from backup

Reviewer: ```suggestion
In the event of a storage failure, a $SERVICE_SHORT automatically recovers from a backup
```

> to the point of failure. If the whole availability zone goes down, your $SERVICE_LONGs are recovered in a different zone. In the event of a user error, where a point-in-time recovery needs to be done, you can [create a point-in-time recovery fork][create-fork].

Reviewer: ```suggestion
to the point of failure. If the whole availability zone goes down, your $SERVICE_LONGs are recovered in a different zone. In the event of a user error, you can [create a point-in-time recovery fork][create-fork].
```

## Enable cross-region backup 

<Availability products={['cloud']} price_plans={['enterprise']} />

> For added reliability, you can enable cross-region backup. In this case, you have two identical backups of your $SERVICE_SHORT at any time, but one of them is in a different AWS region.

Reviewer: ```suggestion
For added reliability, enable cross-region backup. This protects your data when an entire AWS region goes down. This means you have two identical backups of your $SERVICE_SHORT at any time, but one of them is in a different AWS region. A cross-region backup is an identical replica of your $SERVICE_SHORT in a different AWS region. Cross regions backups are updated daily and weekly in the same way as a regular backup. You can have one cross-region backup for a $SERVICE_SHORT.  
```

Author: Cross-region backups are identical replicas of your $SERVICE_SHORT, but one of them is in a different AWS region --> this reads like you have multiple cross-region backups per service, which is not the case. Also I wanted to clarify that you don't create cross-region backup once and it stays the same - it gets updated daily and weekly just like the regular backup, so at any point in time you have two identical ones - cross and same-region.

> You enable cross-region backup when creating a $SERVICE_SHORT, or configure it for an existing $SERVICE_SHORT in $CONSOLE_LONG:

Reviewer: ```suggestion
You enable cross-region backup when you create a $SERVICE_SHORT, or configure it for an existing $SERVICE_SHORT in $CONSOLE_LONG:
```

<Procedure>

1. In [$CONSOLE_SHORT][console], select your $SERVICE_SHORT and click `Operations` > `Backup & restore`. 

1. In `Cross-region backup`, select the region in the dropdown and click `Enable backup`. 

   ![Create cross-region backup](https://assets.timescale.com/docs/images/tiger-cloud-console/create-cross-region-backup-tiger-cloud.png)

   You can now see the backup, its region, and creation date in a list. 

</Procedure>

You can have one cross-region backup per $SERVICE_SHORT. To change the region of your backup:

<Procedure>

1. In [$CONSOLE_SHORT][console], select your $SERVICE_SHORT and click `Operations` > `Backup & restore`.

1. Click the trash icon next to the existing backup to disable it. 

   ![Disable cross-region backup](https://assets.timescale.com/docs/images/tiger-cloud-console/cross-region-backup-list-tiger-cloud.png)

1. Create a new backup in a different region. 

</Procedure>

## Create a point-in-time recovery fork

> You recover your $SERVICE_SHORT from a destructive or unwanted action by creating a point-in-time recovery fork. You can recover a $SERVICE_SHORT to any point within the period [defined by your pricing plan][pricing-and-account-management]. The original $SERVICE_SHORT stays untouched to avoid losing data created since the time of recovery.

Reviewer: ```suggestion
To recover your $SERVICE_SHORT from a destructive or unwanted action, create a point-in-time recovery fork. You can recover a $SERVICE_SHORT to any point within the period [defined by your pricing plan][pricing-and-account-management]. The original $SERVICE_SHORT stays untouched to avoid losing data created since the time of recovery.
```

Since the point-in-time recovery is done in a fork, to migrate your
application to the point of recovery, change the connection
strings in your application to use the fork. The provision time for the
recovery fork is typically less than twenty minutes, but can take longer
depending on the amount of WAL to be replayed.

To avoid paying for compute for the recovery fork and the original $SERVICE_SHORT, pause the original to only pay storage costs.

You initiate a point-in-time recovery from a same-region or cross-region backup in $CONSOLE_LONG:

<Tabs label="Point-in-time recovery in Tiger Cloud Console">

<Tab title="Same-region backup">

<Procedure>

1.  In [$CONSOLE][console], from the `Services` list, ensure the $SERVICE_SHORT
    you want to recover has a status of `Running` or `Paused`.
1.  Navigate to `Operations` > `Backup & restore` and click `Create recovery fork`. 
1.  Select the recovery point, ensuring the correct time zone (UTC offset).
1.  Configure the fork. 

    You can add an HA replica, enable cross-region backup, change the compute resources, and

>     add a connection pooler as part of this process. It is recommended to match

Reviewer: ```suggestion
    add a connection pooler as part of this process. Best practice is to match
```
